app/code/community/Bolt/Boltpay/Model/ShippingAndTax.php

```diff
@@ -219,6 +220,15 @@ public function applyShippingRate($quote, $shippingRateCode) {
         $shippingAddress = $quote->getShippingAddress();
 
         if (!empty($shippingAddress)) {
+
+            Mage::dispatchEvent(
+                'bolt_boltpay_shipping_method_applied_before',
+                array(
+                    'quote'=> $quote,
+                    'shippingMethodCode' => $shippingRateCode
```

Comment:
passing string field as structured data seems error prone. how about we instead pass Rate object?